src/core/mount.c

```diff
@@ -164,6 +164,11 @@ static bool mount_needs_quota(const MountParameters *p) {
         if (p->fstype && fstype_is_network(p->fstype))
                 return false;
 
+        /* xfs uses journaling to keep quota information and thus provide consistency. It also has quota accounting 
+         * turned on at mount time and enable limit enforcement by default. */
```

Comment:
Unfortunately, the situation is not that simple.
The reply from our quota kernel developer is:

-->
1) quotacheck needs to be run for some filesystems after they are mounted
if the filesystem was not unmounted cleanly.

2) You may need to run quotaon to enable quota usage tracking and/or
enforcement.

Sadly the situation is rather complex due to historical burden. So let me
go here through filesystems supporting quotas:

ext2 - needs both 1) and 2)
ext3 - if configured using usrjquota/grpjquota mount options, needs only 2)
ext4 - here configuration matters a lot. If the ext4 filesystem is created
  without a journal, it needs 1). If the ext4 filesystem is created without
  QUOTA filesystem feature (see ext4(5)), it needs 2). Otherwise everything
  is done automatically (which is the desired configuration but there are
  lots of historical configurations around).
reiserfs - needs 2).
jfs - needs 2)
f2fs - if configured using usrjquota/grpjquota/prjjquota mount options, it
  needs 2). Otherwise everything is done automatically. I don't think
  usrjquota/grpjquota/prjjquota got used much in the wild.
xfs - nothing needed
gfs2 - nothing needed
ocfs2 - nothing needed
--<

So, it can't be easily done with inverting the patch.

Reply:
Could you mention that in the code (with references if possible)?

Reply:
Thanks.